xds/src/main/java/io/grpc/xds/XdsClientImpl.java

```diff
@@ -708,22 +710,30 @@ private void handleLdsResponseForListener(DiscoveryResponse ldsResponse) {
           ldsResponse.getVersionInfo(), "Malformed LDS response: " + e);
       return;
     }
-    adsStream.sendAckRequest(ADS_TYPE_URL_LDS, ImmutableList.<String>of(),
-        ldsResponse.getVersionInfo());
     if (requestedListener != null) {
       if (ldsRespTimer != null) {
         ldsRespTimer.cancel();
         ldsRespTimer = null;
       }
-      ListenerUpdate listenerUpdate = ListenerUpdate.newBuilder()
-          .setListener(EnvoyServerProtoData.Listener.fromEnvoyProtoListener(requestedListener))
-          .build();
-      listenerWatcher.onListenerChanged(listenerUpdate);
+      try {
+        ListenerUpdate listenerUpdate = ListenerUpdate.newBuilder()
+            .setListener(EnvoyServerProtoData.Listener.fromEnvoyProtoListener(requestedListener))
+            .build();
+        listenerWatcher.onListenerChanged(listenerUpdate);
```

Comment:
The ACK should be sent before the information is used.

Reply:
Okay, I see what you mean. I'll need to restructure the code so I call "onListernerChanged" only after sending the ACK and if there is an exception then send a NACK instead.

Reply:
fixed

Reply:
Should also include `onResourceDoesNotExit()`.